r/R/compute.R

```diff
@@ -284,3 +284,14 @@ cast_options <- function(safe = TRUE, ...) {
   )
   modifyList(opts, list(...))
 }
+
+#' @export
+strsplit <- function(x, split, fixed = TRUE, perl = FALSE, useBytes = FALSE, ...) UseMethod("strsplit")
+
+#' @export
+strsplit.default <- base::strsplit
+
+#' @export
+strsplit.ArrowDatum <- function(x, split, fixed = TRUE, perl = FALSE, useBytes = FALSE, ...){
```

Comment:
I think we should use `fixed = FALSE` for consistency with `base::strsplit()`. Then the code here should throw an error unless `fixed` is `TRUE`. So the user will need to specify `fixed = TRUE` for it to work.

Reply:
`fixed = FALSE` doesn't work? Is that a separate regex version, not yet implemented in C++? (If so, please make a jira for it).

Assuming that's the case, can we still support `fixed = FALSE` as long as `split` is not a regexp special character? For example, `strsplit("The quick brown fox", " ")` is identical whether or not you do `fixed = TRUE` or `FALSE`.

Reply:
Yep, the splitting on a regex pattern is not implemented in C++. `split_pattern` was added in https://issues.apache.org/jira/browse/ARROW-9991 and it's strictly for literal strings; there's no RE2 there.

>as long as split is not a regexp special character

The logic to detect that a character string is not a regular expression could get very, very complicated. But I suppose it would be straightforward to detect some very obvious cases of not-a-regex. @thisisnic I'm happy to help with this if you need a hand!

Reply:
Per R's docs, 

> The metacharacters in extended regular expressions are `. \ | ( ) [ { ^ $ * + ?`

So we could check for those.

Reply:
@thisisnic could you check if there's already an open Jira for adding a C++ compute function to split a string by a regular expression, and if not please add one? Thanks!

Reply:
Updated this and added JIRA ticket for regex splitting

Reply:
Thanks! It's ARROW-12608